code/modules/mob/living/simple_animal/morph/morph.dm

```diff
@@ -0,0 +1,179 @@
+							You may take the form of anything nearby by shift-clicking it. This process will alert any nearby \
+							observers, and can only be performed once every five seconds. While morphed, you move faster, but do \
+							less damage. In addition, anyone within three tiles will note an uncanny wrongness if examining you. \
+							You can attack any item or dead creature to consume it - creatures will fully restore your health. \
+							Finally, you can restore yourself to your original form while morphed by shift-clicking yourself.</b>"
+
+/mob/living/simple_animal/hostile/morph/examine(mob/user)
+	if(morphed)
+		form.examine(user) // Refactor examine to return desc so it's static? Not sure if worth it
+		if(get_dist(user,src)<=3)
+			user << "<span class='warning'>It doesn't look quite right...</span>"
+	else
+		..()
+	return
+
+/mob/living/simple_animal/hostile/morph/proc/allowed(atom/movable/A) // make it into property/proc ? not sure if worth it
+	if(istype(A,/obj/screen))
+		return 0
+	if(istype(A,/obj/singularity))
+		return 0
+	if(istype(A,/mob/living/simple_animal/hostile/morph))
+		return 0
+	return 1
+
+/mob/living/simple_animal/hostile/morph/proc/eat(atom/movable/A)
+	if(A && A.loc != src)
+		visible_message("<span class='warning'>[src] swallows [A] whole!</span>")
+		A.forceMove(src)
+		return 1
+	return 0
+
+/mob/living/simple_animal/hostile/morph/ShiftClickOn(atom/movable/A)
+	if(morph_time <= world.time && !stat)
+		if(A == src)
+			restore()
+			return
+		if(istype(A) && allowed(A))
+			assume(A)
+	else
+		src << "<span class='warning'>Your chameleon skin is still repairing itself!</span>"
+		..()
+
+/mob/living/simple_animal/hostile/morph/proc/assume(atom/movable/target)
+	morphed = 1
+	form = target
+
+	appearance = target.appearance
+	transform = initial(transform)
+	pixel_y = initial(pixel_y)
+	pixel_x = initial(pixel_x)
+	visible_message("<span class='warning'>[src] suddenly twists and changes shape, becoming a copy of [target]!</span>", \
+					"<span class='notice'>You twist your body and assume the form of [target].</span>")
+	//Morphed is weaker
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	speed = 0
+
+	morph_time = world.time + MORPH_COOLDOWN
+	return
+
+/mob/living/simple_animal/hostile/morph/proc/restore()
+	if(!morphed)
+		return
+	morphed = 0
+	form = null
+
+	//anim(loc,src,'icons/mob/mob.dmi',,"morph",,src.dir)
+
+	visible_message("<span class='warning'>[src] suddenly collapses in on itself, dissolving into a pile of green flesh!</span>", \
+					"<span class='notice'>You reform to your normal body.</span>")
+	name = initial(name)
+	icon = initial(icon)
+	icon_state = initial(icon_state)
+	overlays.Cut()
+
+	//Baseline stats
+	melee_damage_lower = initial(melee_damage_lower)
+	melee_damage_upper = initial(melee_damage_upper)
+	speed = initial(speed)
+
+	morph_time = world.time + MORPH_COOLDOWN
+
+/mob/living/simple_animal/hostile/morph/handle_state_icons()
```

Comment:
If I don't have this system here, then the morph will set its `icon_state` when polymorphed to something else, back to its `icon_living` condition. Since most spritesheets lack a "morph" sprite, this results in the mob being invisible, which is obviously bad.